packages/test/service-load-test/src/faultInjectionDriver.ts

```diff
@@ -0,0 +1,173 @@
+    IDocumentService,
+    IDocumentServiceFactory,
+    IDocumentStorageService,
+    IResolvedUrl,
+} from "@fluidframework/driver-definitions";
+import {
+    IClient,
+    ISummaryTree,
+    IErrorTrackingService,
+    IDocumentMessage,
+    INack,
+    NackErrorType,
+} from "@fluidframework/protocol-definitions";
+
+export class FaultInjectionDocumentServiceFactory implements IDocumentServiceFactory {
+    private  readonly _last = new Map<IResolvedUrl, FaultInjectionDocumentService>();
+
+    public get protocolName() { return this.internal.protocolName;}
+    public get documentServices() {return this._last;}
+
+    constructor(private readonly internal: IDocumentServiceFactory) { }
+
+    async createDocumentService(resolvedUrl: IResolvedUrl, logger?: ITelemetryBaseLogger): Promise<IDocumentService> {
+        const internal = await this.internal.createDocumentService(resolvedUrl, logger);
+        const ds = new FaultInjectionDocumentService(internal);
+        assert(!this._last.has(resolvedUrl), "one ds per resolved url instance");
+        this._last.set(resolvedUrl, ds);
+        return ds;
+    }
+    async createContainer(
+        createNewSummary: ISummaryTree,
+        createNewResolvedUrl: IResolvedUrl,
+        logger?: ITelemetryBaseLogger):
+        Promise<IDocumentService> {
+        return this.internal.createContainer(
+            createNewSummary,
+            createNewResolvedUrl,
+            logger,
+        );
+    }
+}
+
+export class FaultInjectionDocumentService implements IDocumentService {
+    private  _lastDeltaStream: FaultInjectionDocumentDeltaConnection | undefined;
+
+    constructor(private readonly internal: IDocumentService) {
+    }
+
+    public get resolvedUrl() {return this.internal.resolvedUrl;}
+    public get policies() {return this.internal.policies;}
+    public get documentDeltaConnection() {
+        return this._lastDeltaStream;
+    }
+
+    async connectToStorage(): Promise<IDocumentStorageService> {
+        return this.internal.connectToStorage();
+    }
+
+    async connectToDeltaStorage(): Promise<IDocumentDeltaStorageService> {
+        return this.internal.connectToDeltaStorage();
+    }
+
+    async connectToDeltaStream(client: IClient): Promise<IDocumentDeltaConnection> {
+        assert(
+            this._lastDeltaStream?.closed !== false,
+            "Document service factory should only have one open connection");
+        const internal = await this.internal.connectToDeltaStream(client);
+        this._lastDeltaStream = new FaultInjectionDocumentDeltaConnection(internal);
+        return this._lastDeltaStream;
+    }
+
+    /**
+     * Returns the error tracking service
+     */
+    getErrorTrackingService(): IErrorTrackingService | null {
+        return this.internal.getErrorTrackingService();
+    }
+}
+
+export interface FaultInjectionDocumentDeltaConnectionEvents extends IEvent{
+    (event: "sumbit" | "submitSignal",listener: () => void);
```

Comment:
```suggestion
    (event: "submit" | "submitSignal",listener: () => void);
```

Reply:
Wait, this interface isn't used, you're (correctly to me) reusing IDocumentDeltaConnectionEvents. So I think just delete this interface right?

Reply:
not used anymore